Library/DiscUtils.Xfs/Directory.cs

```diff
@@ -95,7 +95,7 @@ private void AddLeafDirExtentEntries(IList<Extent> extents, Dictionary<string, D
             {
                 if (extent.StartOffset < leafOffset)
                 {
-                    for (int i = 0; i < extent.BlockCount; i++)
+                    for (long i = 0; i < extent.BlockCount; i++)

```

Comment:
My test and the documentation show the values 1 & 2. I've changed this to long to prevent a pissible overflow in the next line `i* Context.SuperBlock.DirBlockSize`.